### PR TITLE
Qbft rework artifacts

### PR DESCRIPTION
**Pull request by rain-on**
## Changelog

- [ ] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

File: consensus/qbft/src/main/java/org/hyperledger/besu/consensus/qbft/statemachine/QbftRound.java [hunk 1]
```diff
@@ -102,26 +100,28 @@ public void createAndSendProposalMessage(final long headerTimeStampSeconds) {
...
   public void startRoundWith(
-      final RoundChangeMetadata roundChangeMetadata, final long headerTimestamp) {
-    final Optional<Block> bestBlockFromRoundChange = roundChangeMetadata.getBlock();
+      final RoundChangeArtifacts roundChangeArtifacts, final long headerTimestamp) {
+    final Optional<PreparedCertificate> peerWasPrepared =
+        roundChangeArtifacts.getBestPreparedPeer();
+
     Block blockToPublish;
-    if (bestBlockFromRoundChange.isEmpty()) {
+    if (peerWasPrepared.isEmpty()) {
```
Comment (jframe) [inline consensus/qbft/src/main/java/org/hyperledger/besu/consensus/qbft/statemachine/QbftRound.java:108]: Don't like this variable name, it implies a boolean to me. Perhaps bestPreparedCertificate?

File: consensus/qbft/src/main/java/org/hyperledger/besu/consensus/qbft/statemachine/RoundChangeArtifacts.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+import java.util.stream.Collectors;
+
+public class RoundChangeArtifacts {
+
+  private final List<SignedData<RoundChangePayload>> roundChanges;
+  private final Optional<PreparedCertificate> bestPreparedPeer;
+
+  public RoundChangeArtifacts(
+      List<SignedData<RoundChangePayload>> roundChanges,
+      Optional<PreparedCertificate> bestPreparedPeer) {
```
Comment (jframe) [inline consensus/qbft/src/main/java/org/hyperledger/besu/consensus/qbft/statemachine/RoundChangeArtifacts.java:34]: nit: final

Comment (rain-on) [reply consensus/qbft/src/main/java/org/hyperledger/besu/consensus/qbft/statemachine/RoundChangeArtifacts.java]: done

**Review by jframe — APPROVED** [review]
LGTM

**Pull request merged**